sdk/identity/azure-identity/src/default_azure_credential.cpp

```diff
@@ -42,12 +43,14 @@ DefaultAzureCredential::DefaultAzureCredential(
   auto const wiCred = std::make_shared<WorkloadIdentityCredential>(options);
   auto const azCliCred = std::make_shared<AzureCliCredential>(options);
   auto const managedIdentityCred = std::make_shared<ManagedIdentityCredential>(options);
+  auto const pipelineCred = std::make_shared<AzurePipelinesCredential>();
```

Comment:
Please revert this change along with the changes to `AzurePipelinesCredential` public surface area. This credential is **by design** not part of the DAC across all languages, and it isn't intended to be.

If you could summarize the problem you are trying to solve here that required such a change, I can try and suggest alternative solutions. I recommend seeing what other languages are doing to get inspiration.

Reply:
The credential should be part of the DAC. Its the preferred credential for Azure pipelines as part of SFI. Not having it in the DAC really brings to light that no one should use DAC.

Reply:
Did this change occur recently, across all languages that implemented the AzurePipelinesCredential? Please share more details of this design discussion and conclusion, offline. If there is a documented doc that describes the motivation behind this change, let me know as I would love to read up on that.

Reply:
Here's the design gist that the languages followed:
https://gist.github.com/KarishmaGhiya/81ee6265ab6e9109d3bf510678878b34/